### PR TITLE
Add function to simulate SiPM charge fluctuations.

### PR DESCRIPTION
**Pull request by paolafer**
This PR encapsulates in a function the code previously added to the script for coincidence reconstruction. This way, the function can be used elsewhere.  A test has been added and the script has been changed to call the function.

### CODE REVIEW

File: antea/scripts/reconstruct_coincidences.py [hunk 1]
```diff
@@ -103,30 +106,8 @@
...
-        # fluctuate charge according to charge resolution
-        sum_sns = evt_sns.groupby(['event_id','sensor_id'])[['charge']].sum()
-        sum_sns = sum_sns.reset_index()
-        charges = sum_sns.charge.values
-        sipmrd  = charges[:, np.newaxis] # to use IC functions, that expect a waveform
-
-        pe_resolution = DataSiPM_idx.Sigma / DataSiPM_idx.adc_to_pes
-        touched_sipms = sum_sns.sensor_id.values
-        pe_res        = pe_resolution[touched_sipms]
-        if len(pe_res) != len(sipmrd) :
```
Comment (carmenromo) [inline antea/scripts/reconstruct_coincidences.py:117]: This condition is not included in the new function. Do you think it is not necessary?

Comment (paolafer) [reply antea/scripts/reconstruct_coincidences.py:117]: I think it is fulfilled by construction...

Comment (carmenromo) [reply antea/scripts/reconstruct_coincidences.py:117]: yeah, I think you are right... I was thinking about a case it isn't fulfilled but I don't find no one...

File: antea/mcsim/sensor_functions_test.py [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    DataSiPM_idx = DataSiPM.set_index('SensorID')
+
+    PATH_IN       = os.path.join(ANTEADATADIR, 'full_body_1ev.h5')
+    sns_response  = load_mcsns_response(PATH_IN)
+    events        = sns_response.event_id.unique()
+
+    fluct_sns_response = apply_charge_fluctuation(sns_response, DataSiPM_idx)
+
+    original   = sns_response.groupby(['event_id']).sensor_id.nunique()
+    fluctuated = fluct_sns_response.groupby(['event_id']).sensor_id.nunique()
```
Comment (carmenromo) [inline antea/mcsim/sensor_functions_test.py:24]: This is a super minor comment, but I would align the two groupbies :)

**Comment by carmenromo** [comment]
The new function does its job and returns a dataframe with the information of the sensors after applying the charge fluctuation. Also a test is added to ensure that the returned df constains the same sensor ids.
Approved!

File: antea/mcsim/sensor_functions_test.py (hunk 2 above)
Comment (paolafer) [reply antea/mcsim/sensor_functions_test.py:24]: Done.

**Pull request merged**